 # Lab 3 - The Pendulum I

In [25]:
# Make sure to run this cell before running any other code cells
%reset -f
import numpy as np
import data_entry2

***Edit the following cell to provide the information requested:***

- Your name: Mary Hosseini
- Your student number: 41004235
- Partner's name: Vanessa Lau

## Part A - Intro and announcements

***TA grading feedback here:***
- Feedback: "In the future please further justify your uncertainties (why you have this confidence interval) and double check your significant figures!"

We will make sure to include why we chose those certain confidence intervals and double check our significant figures :) 

## Part B - Invention Activity

***Notes from the invention activity. Make sure to include a formula for the new statistical tool and a description of how to use and interpret it.*** 



- Uncertainty used to make comparisons (from experimental steps this is when you would "compare with model or other data" in other words, comparing your results with some other results)
- We worked through our understandings of what we had first. This is explained in the student solution section below

### Student Solution
- We came up with $\frac{A-B}{|u_A^2| + |u_B^2|}$. We chose A-B to be because we know we needed the difference between the two distributions, we also knew that our confidence would increase as the distance between the curves increased, hence why we put it up top. We then decided to take the absolute value of the uncertainties then add them together and put it in the denomenator so that we could accurately calculate our confidence between 1-4. This however was not successful because we would get really high or really low values and nothing in between. 


### Instructor Solution:
- Introduction: We take the absolute value of $A-B$ to ensure that we will always have positive distribution difference. This solution however doesn't account for the uncertainty
- Solution two (t-score The correct solution): We take the absolute value of $A-B$ to keep the difference between the distributions positive. We then divide that by the square root of $u_A^2 + u_B^2$. This way we can be confident in the two distributions when we have a big value for the difference between distributions and/or when we have a small amount of uncertainty. 
- We call this a t-score value. The closer the t-score is to 5, the more confidence you have that the results are different for that given relative uncertainty. The closer the t-score is to 1, you are not confident at all that that the results are different. If the t-score is closer to 
- A higher t-score with a lower relative uncertainty, the higher is your confidence in different results. A high relative uncertainty and a high t-score, means there's a difference with results with low reliability. A low t-score with a low relative uncertainty, means you have low confidence but the data has high reliability. Lastly a low confidence occurs when you have high uncertainty and low t-score, in this case you are not confident because there's low reliability and difference. 

## Reporting t-scores
1) Report t-score value
2) Confidence
3) Report relative uncertainty
- Provided with examples in class: "We are extremely confident there is a difference because our t-score was 7.4, this is a very reliable measurement because we had really low relative uncertainties of 0.001 and 0.004, much lower than other groups."

## Examples Calculations:
- First cell is Pair One t-score and second cell is Pair Two t-score

In [26]:
np.abs(45-35)/ np.sqrt(15**2 + 15**2)

0.4714045207910317

In [27]:
np.abs(20-60)/ np.sqrt(5**2 + 5**2)

5.65685424949238

## Part C - Start of familiarization

***Discuss the experimental goal and familiarization done in Part C of the student instructions. You can also add any observations from your prelab interactions with the simulation here. Write down a measurement plan for your first round of data collection.***

The experimental goal is to determine if the period of a pendulum depends on amplitude by comparing the period with initial release angles of 10° and 20°. From the measurements done in the prelab and from comparing prelab results with other students, we think that the period of a pendulum does not depend on the amplitude because the angle difference is so small. 

### Measurement Plan
We will use a UBC student card to make sure the string of the pendulum lined up with the correct angle we are looking for. For example when we want our pendulum to start from 15 degree. We lined up the ubc card with the correct angle first then we pull the pendulum until it was correctly lined up with the card. After which we release the pendulum and start the timer to measure the pendulum doing 10 cycles (10 times of it moving back and forth to the same spot). 

### Comparison to Stimulation
The stimulation was a lot easier and took less time to line up. In real life preparation for the pendulum to be correctly lined up and ready to be let go took around 2-3 minutes. Another difference is that the real life pendulum needed to be lined up using two people, it would be more difficult to perform this with one person. 

## Part D - Class discussion of pre-lab results

- From the example results, we can see that as we do more trails, the uncertainty decreases therefore reduce uncertainty with doing more cycles and trials
- Average Period formula: $T_{Ave} = \frac{t_{ave}}{M_{cycles}}$
- Uncertainty from the example results: relative uncertainty decreases so the confidence increases
- Uncertainty formula: $u_{T_{Ave}} = \frac{u_{t_{ave}}}{M_{cycles}}$
- Aim should be to be able to get an accurate t-score with low uncertainty so that we can be confident in our results

## Part E - Measurements and Analysis

***Write some notes below detailing your initial measurement plan***

We used a UBC student card to make sure the string of the pendulum lined up with the correct angle we were looking for. For example when we wanted our pendulum to start from 15 degree. We lined up the ubc card with the correct angle first then we pulled the pendulum until it was correctly lined up with the card. After which we released the pendulum. 

We repeated this for a total of 10 trials and we also measured for the pendulum to go for 10 cycles before we stopped the timer. 

We chose 10 cycles because we saw from the results displayed in the presentation that the higher the cycles the lower the uncertainty. 

***Use the cell below to load in a blank spreadsheet to fill in with your initial round of data. (Note that you should have 2 columns of data, of equal length.)***

In [28]:
de1 = data_entry2.sheet("lab03_1")

Sheet name: lab03_1.csv


***Use the cell below for your analysis code. Insert additional cells below as needed. (Note that you can copy in and update your analysis code from the prelab.)***

In [29]:
# For Angle = 10
t1Ave = np.mean(Time10Vec)
print("Average of t1 is", t1Ave)
ut1Ave = np.std(Time10Vec, ddof=1)
print("Standard deviation of t1 is", ut1Ave)

T1Ave = np.mean(Period10Vec/10)
print("Average of period1 is", T1Ave)
uT1Ave = np.std(Period10Vec/10, ddof=1)
print("Standard deviation of period1 is", uT1Ave)

t1Count = Time10Vec

relt1ave = ut1Ave/len(t1Count)
print("Relative uncertainty of t1 is", relt1ave)
relT1ave = uT1Ave/len(t1Count)
print("Relative uncertainty of period1 is", relT1ave)

# For Angle = 20
t2Ave = np.mean(Time20Vec)
print("Average of t1 is", t2Ave)
ut2Ave = np.std(Time20Vec, ddof=1)
print("Standard deviation of t1 is", ut2Ave)

T2Ave = np.mean(Period20Vec/10)
print("Average of period1 is", T2Ave)
uT2Ave = np.std(Period20Vec/10, ddof=1)
print("Standard deviation of period2 is", uT2Ave)

t2Count = Time20Vec

relt2ave = ut1Ave/len(t2Count)
print("Relative uncertainty of t2 is", relt2ave)
relT2ave = uT1Ave/len(t2Count)
print("Relative uncertainty of period2 is", relT2ave)

# t-score value
tScore = np.abs(T2Ave-T1Ave)/np.sqrt(uT2Ave**2+uT1Ave**2)
print("The t-score value is", tScore)

Average of t1 is 18.517000000000003
Standard deviation of t1 is 0.11106054404893045
Average of period1 is 0.18517
Standard deviation of period1 is 0.0011106054404893045
Relative uncertainty of t1 is 0.011106054404893045
Relative uncertainty of period1 is 0.00011106054404893046
Average of t1 is 18.58
Standard deviation of t1 is 0.11498792207106913
Average of period1 is 0.18580000000000002
Standard deviation of period2 is 0.001149879220710693
Relative uncertainty of t2 is 0.011106054404893045
Relative uncertainty of period2 is 0.00011106054404893046
The t-score value is 0.3940836756245743


***Reflect on the results of your first round below***

The t-score that we got is relatively low (0.39408) and our relative uncertainty was 0.00011 and 0.00011, so we are not confident that the values are different. Other groups had a higher relative uncertainty 0.05900 and 0.02900, and a lower t-score of 1.8000 whereas ours was much lower so we know that our results have more reliable measurements. 

***Iterative cycles of improvements if your have time, or come up with an improvement for Lab 4***

For the next lab a strategy that we could implement is increasing the M and increasing the number of trials so that we could get a more accurate result. By increasing the number of trials we can improve accuracy, and by increasing M, we can decrease relative uncertainty. We could also line up the pendulum more accurately with the angle ruler so that we could get more accurate and precise results. 

## Part F - COMPLETE YOUR SUMMARY OF RESULTS AND SUBMIT

### A brief summary of your key results and key comparisons in 1-2 sentences.

The t-value that we got is 0.39408 with relative uncertainty of 0.00011 and 0.00011, however we are not confident that the Period values are different. Other groups had higher relative uncertainties of 0.05900 and 0.02900 and lower t-score of 1.8000 and since ours had lower uncertainty, we are assured that our results have more reliable measurements. 

### 1-2 sentences addressing the experimental goal and describing the conclusions you can make from the results of your comparisons.
The goal of this lab was to determine if the period of a pendulum depends on amplitude. This was done by comparing the period with initial release angles of 10° and 20°. From our results, we can deduce that the period of a pendulum does not depend on the amplitude. Our deduction is based on the fact that we have a low t-score and low relative uncertainties (so we know that our measurements are accurate). 

### 1 sentence containing your thoughts for further investigation. If you were to spend more time with this experiment, what would you plan to do next?

We would increase the number of trials and we would change the number of swings of the pendulum (increase it) so that we can ensure we get more accurate results. 

# Submit

Steps for submission:

1. Click: Run => Run_All_Cells
2. Read through the notebook to ensure all the cells executed correctly and without error.
3. Correct any errors you find.
4. File => Save_and_Export_Notebook_As->HTML
5. Upload the HTML document to the lab submission assignment on Canvas.

In [30]:
# trick to display tables in html output
display_sheets()

Sheet: de1   File: lab03_1.csv


Time15 Period15   Time20 Period20   Time10 Period10
Units  seconds  seconds  seconds  seconds  seconds  seconds
0        18.42    1.842     18.6     1.86    18.71    1.871
1        18.46    1.846    18.33    1.833    18.43    1.843
2        18.41    1.841    18.59    1.859    18.42    1.842
3        18.42    1.842    18.59    1.859    18.57    1.857
4        18.21    1.821    18.72    1.872    18.34    1.834
5        18.32    1.832    18.58    1.858    18.44    1.844
6        18.50    1.850    18.45    1.845    18.58    1.858
7        18.47    1.847    18.65    1.865    18.50    1.850
8        18.39    1.839    18.70     1.87    18.62    1.862
9        18.27    1.827    18.59    1.859    18.56    1.856